src/openswathalgo/source/OPENSWATHALGO/ALGO/Scoring.cpp

```diff
@@ -105,7 +105,14 @@ namespace OpenSwath
       x_len = std::sqrt(x_len);
       y_len = std::sqrt(y_len);
 
-      return std::acos(dotprod / (x_len * y_len));
+      // normalise, avoiding a divide by zero. See unit tests for what happens
+      // when one of the vectors has a length of zero.
+      dotprod /= std::max(std::numeric_limits<double>::min(), x_len * y_len);
```

Comment:
imho this changes the sign of the dotproduct and sets it to -1 or +1.
wouldn't it be easier to just do a simple if (y_len == 0) return ... etc.

Reply:
I agree, one could do something like this, right?

```suggestion
      if (x_len * y_len == 0) {return pi}; 
      return std::acos(dotprod / (x_len * y_len));
```

Reply:
I agree on the `std::max` call here, it definitely doesn't take into account sign and it should.

```suggestion
      double denominator = x_len * y_len;
      dotprod = (denominator == 0.0) ? 0.0 : dotprod / denominator;
```

But you still need the `clip` below to account for floating point errors that can cause `acos` to blow up